spec/rspec/matchers/composable_spec.rb

```diff
@@ -49,6 +49,32 @@ module Matchers
         end
       end
 
+      context "when reusing a matcher that memoizes state" do
```

Comment:
Since you've fixed `contain_exactly`, I believe this example would better go to its spec.